File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/teleport/TeleportService.java
package top.likoslupus.cellulosesz.api.teleport;

import top.likoslupus.cellulosesz.api.platform.CellPlayer;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface TeleportService {

    CompletableFuture<TeleportResult> teleport(
            CellPlayer player,
            CellLocation target,
            TeleportOptions options
    );

    boolean cancelWarmup(UUID uuid, TeleportStatus status);

    boolean warmingUp(UUID uuid);

    CompletableFuture<Void> rememberBackLocation(CellPlayer player);

    CompletableFuture<Void> rememberBackLocation(UUID uuid, CellLocation location);

    Optional<CellLocation> backLocation(UUID uuid);

    void shutdown();

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/teleport/TeleportService.kt
package top.likoslupus.cellulosesz.api.teleport

import top.likoslupus.cellulosesz.api.platform.CellPlayer
import java.util.*
import java.util.concurrent.CompletableFuture

public interface TeleportService {

    public fun teleport(
        player: CellPlayer,
        target: CellLocation,
        options: TeleportOptions
    ): CompletableFuture<TeleportResult>

    public fun cancelWarmup(uuid: UUID, status: TeleportStatus): Boolean

    public fun warmingUp(uuid: UUID): Boolean

    public fun rememberBackLocation(player: CellPlayer): CompletableFuture<Void>

    public fun rememberBackLocation(uuid: UUID, location: CellLocation): CompletableFuture<Void>

    public fun backLocation(uuid: UUID): CellLocation?

    public fun shutdown()

}
